### ISSUE TITLE
Add labs to repo

### ISSUE THREAD

**Issue by ErikBjare** [bjornregnell/lth-eda016-2015#1]
It would be nice to have all the labs in the repo so that people can easily make improvements such as:

 - Clarify the instructions
 - Fix spelling errors
 - Improve the wording of (or even add) explanations

All improvements should be managed via pull requests as usual.

PS. I'm particularly interested in lab 10 named "Life".

**Comment by bjornregnell** [comment]
Labs are hosted in a closed repo on Bitbucket with unclear license - I'll check with the other teachers if it's possible to make them public.

**Comment by ErikBjare** [comment]
That would be great! Thanks :+1:

**Comment by bjornregnell** [comment]
It is summer break so I don't expect rapid response from my colleagues, but as it is lab-life that was of particular interest, I'll upload the latex body for that lab (it's part of a complex build system so don't expect to be able to generate any pdf from it, but at least you can edit and create a pull request...). Thanks for your efforts; any improvement proposal / typo corrections etc. welcome!

**Comment by bjornregnell** [comment]
Now available here:
https://github.com/bjornregnell/lth-eda016-2015/blob/master/labs/lab-life/instruktioner/uppgift.tex
It is just a copied file from another repo, so I'd have to copy back manually for now. (Decision pending on making labs public, but let me know if you spot anything that needs to be updated.) The pre-print of the compendium must be made first week of august...

**Comment by ErikBjare** [comment]
Great!

I have to admit this idea was part of a larger plan to try and make some of the course participants (and lab/exercise-assistants) engage with building/improving the course by giving concrete examples for improvements (sent via PRs). The Life lab looks good, can't find any obvious issues.

Since Life is strongly related to the field [cellular automata]() (which a new student could easily navigate with considerable ease and even implement many of the ideas) it would be nice to have a list of "Read more" links on the bottom of the assignment for further study.

Take for example this [cyclic cellular automaton](https://en.wikipedia.org/wiki/Cyclic_cellular_automaton):

![Cyclic automaton](https://upload.wikimedia.org/wikipedia/commons/8/8d/Super_cyclic.gif)

It's easy to implement, but provides a completely different behavior with a relatively simple change of rules.

Most students might not care, but if even a few do that's of considerable worth. If you are interested I could compile a list of related (and interesting) things. Some of which can be easily implemented once the student has made a solution to the exercise.

Incomplete list:

 - [Cyclic cellular automatons](https://en.wikipedia.org/wiki/Cyclic_cellular_automaton) - Can be implemented on top of the solution
 - [Wireworld](https://en.wikipedia.org/wiki/Wireworld) - Can be implemented on top of the solution

\<anti-java-bias>
It would be nice if we could have interactive labs like this one written in something that's more easily shared, like JavaScript on a HTML canvas. I have a Life implementation [here](http://erik.bjareholt.com/GameOfLife/) for example.
\</anti-java-bias>

I might compliment this more later on, but I need to change location right now so will be done later if so.

**Comment by bjornregnell** [comment]
Nice with larger plans ;) I have one larger plan of my own: Is I'm taking over the responsibility of the beginner programming course, I'll use the first year to learn how the pedagogic (what you learn, e.g. simple vector/matrix algorithms) is connected to the current activation+examination (e.g. a lab on game of life) and then the second year change things that have a large potential to improve student learning in the direction that I would like to take us. I'm playing with the idea to move to a Scala+Java8 combo and involve as many as possible in creating open source teaching material during this study year aiming at going live in August 2016, but that is perhaps more of a dream than a plan ;). I'd be happy to talk about such ideas when we meet live! For now I'm closing this issue; I'd be very happy to integrate a pull request for lab-life with a new \subsection{Extra uppgifter och fördjupningslänkar} by the end of the text that provides some bullets on ideas of further implementation quests and further reading etc.

**Comment by bjornregnell** [comment]
I've sent a query to my colleagues, but I guess we will discuss this later in August.

**Comment by bjornregnell** [comment]
I'm happy to accept errata and suggestions, but I want to avoid multiple repos and will keep the lab in orig. repo for now. (In the future we might move the labs to GitHub.) I'm closing this issue for now.
